exporter/instanaexporter/exporter.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2022, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package instanaexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/instanaexporter"
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"runtime"
+	"strings"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/consumer/consumererror"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/instanaexporter/internal/backend"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/instanaexporter/internal/converter"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/instanaexporter/internal/converter/model"
+)
+
+type instanaExporter struct {
+	config    *Config
+	client    *http.Client
+	settings  component.TelemetrySettings
+	userAgent string
+}
+
+func (e *instanaExporter) start(_ context.Context, host component.Host) error {
+	client, err := e.config.HTTPClientSettings.ToClient(host, e.settings)
+	if err != nil {
+		return err
+	}
+	e.client = client
+	return nil
+}
+
+func (e *instanaExporter) pushConvertedTraces(ctx context.Context, td ptrace.Traces) error {
+	converter := converter.NewConvertAllConverter(e.settings.Logger)
+	spans := make([]model.Span, 0)
+
+	hostID := ""
+	resourceSpans := td.ResourceSpans()
+	for i := 0; i < resourceSpans.Len(); i++ {
+		resSpan := resourceSpans.At(i)
+
+		resource := resSpan.Resource()
+
+		hostIDAttr, ex := resource.Attributes().Get(backend.AttributeInstanaHostID)
```

Comment:
It took me a while to realize that "ex" probably means "exists". Perhaps use `ok` instead? Or even `exists`?

Reply:
Done